Cargo.toml

```
@@ -78,15 +75,12 @@ harness = false
 name = "circuit"
 harness = false
 
-[[bench]]
-name = "poseidon"
-harness = false
-
 [profile.release]
 debug = true
 
 [profile.bench]
 debug = true
 
 [patch.crates-io]
-halo2 = { git = "https://github.com/zcash/halo2.git", rev = "f9b3ff2aef09a5a3cb5489d0e7e747e9523d2e6e" }
+halo2_proofs = { git = "https://github.com/zcash/halo2.git", branch = "patch-gadgets-import" }
+halo2_gadgets = { git = "https://github.com/zcash/halo2.git", branch = "patch-gadgets-import" }
```

We should never reference a branch, because that can move; instead we should reference a specific revision. Now that https://github.com/zcash/halo2/pull/475 has merged, this should be:
```suggestion
halo2_proofs = { git = "https://github.com/zcash/halo2.git", rev = "f3502421cd48767ff84e2ebd965385fc29c0940e" }
halo2_gadgets = { git = "https://github.com/zcash/halo2.git", rev = "f3502421cd48767ff84e2ebd965385fc29c0940e" }
```